docs/concepts/scale-up.md

```diff
@@ -0,0 +1,47 @@
+Now, it is detected whether peerUrl was TLS enabled or not for single node etcd cluster. Etcd-druid can now use this information to take action:
+- If peerUrl was already TLS enabled then no action is required from etcd-druid side. Etcd-druid can proceed with scaling up the cluster.
+- If peerUrl was not TLS enabled then etcd-druid has to intervene and make sure peerUrl should be TLS enabled first for the single node before marking the cluster for scale-up.
+
+### Action taken by etcd-druid to enable the peerUrl TLS
+1. Etcd-druid will updates the `etcd-bootstrap` config-map with new config like initial-cluster,initial-advertise-peer-urls etc. Backup-restore will detect this change and update the member lease annotation to `member.etcd.gardener.cloud/tls-enabled: "true"`.
+2. In case the peerUrl TLS has been changed to `enabled`: Etcd-druid will adds tasks to the deployment flow.
+    - To ensure that the TLS enablement of peerUrl is properly reflected in etcd, the existing etcd StatefulSet pods should be restarted twice. 
+    - The first restart pushes a new configuration which contains PeerUrlTLS configuration. Backup-restore will update the member peer url. This will result in the change of the peer url in the etcd's database, but it may not reflect in the already running etcd container. Ideally a restart of an etcd container would have been sufficient but currently k8s doesn't expose an API to force restart a single container within a pod. Therefore, we need to restart the StatefulSet pod(s) once again. When the pod(s) is restarted the second time it will now start etcd with the correct peer url which will be TLS enabled.
+    - To achieve 2 restarts following is done:
+        * An update is made to the spec mounting the peer URL TLS secrets. This will cause a rolling update of the existing pod.
+        * Once the update is successfully completed, then we delete StatefulSet pods, causing a restart by the StatefulSet controller.
+
+
+### After PeerUrl is TLS enabled
+After peerUrl TLS enablement for single node etcd cluster, now etcd-druid adds a scale-up annotation: `gardener.cloud/scaled-to-multi-node` to etcd statefulset and etcd-druid will patches the statefulset `.spec.replicas` to `3`(say), now statefulset controller will bring up new pods(etcd with backup-restore as a sidecar). Now etcd's sidecar i.e backup-restore will check whether this member is already a part of cluster or not and incase it is unable to check (may be due to some network issues) then backup-restore checks presence of this annotation: `gardener.cloud/scaled-to-multi-node` in etcd statefulset to detect scale-up if it finds out it is the scale-up case then backup-restore adds new etcd member as a [learner](https://etcd.io/docs/v3.3/learning/learner/) first then start the etcd learner by providing the correct configuration. Once learner is get in sync with etcd cluster leader, then learner will get promoted to voting member.
```

Comment:
```suggestion
After peerUrl TLS enablement for single node etcd cluster, etcd-druid adds a scale-up annotation: `gardener.cloud/scaled-to-multi-node` to the etcd statefulset and will patch the statefulsets `.spec.replicas` to `3`(for example). The statefulset controller will then bring up new pods(etcd with backup-restore as a sidecar). Now etcd's sidecar i.e backup-restore will check whether this member is already a part of a cluster or not and incase it is unable to check (may be due to some network issues) then backup-restore checks presence of this annotation: `gardener.cloud/scaled-to-multi-node` in etcd statefulset to detect scale-up. If it finds out it is the scale-up case then backup-restore adds new etcd member as a [learner](https://etcd.io/docs/v3.3/learning/learner/) first and then starts the etcd learner by providing the correct configuration. Once the learner gets in sync with the etcd cluster leader, it will get promoted to a voting member.
```